source/linux/Foundational_Components_Multimedia_wave5.rst

```diff
@@ -630,27 +987,37 @@ Buffer import on decoder can be tested by setting the ``capture-io-mode`` to ``5
 Configuration of CMA Size
 *************************
 
-The CMA size can be increased or decreased depending on the requirement and the memory map usage by other components.
+Wave5 uses common CMA region to read and write buffers to, since the IP does not posses an on-chip MMU.
 
 .. ifconfig:: CONFIG_part_variant in ('J721S2')
 
-   The macro that specifies the CMA size is CONFIG_CMA_SIZE_MBYTES present in the file :file:`arch/arm64/configs/tisdk_j721s2-evm_defconfig` in the linux directory of sdk. The default value is 896MB.
+   The CMA size can be adjusted by opening the board dtsi file :file:`arch/arm64/boot/dts/ti/k3-j721s2-som-p0.dtsi`,
+   finding the "linux,cma" node and changing the size attribute. The default value is 896MB.
 
 .. ifconfig:: CONFIG_part_variant in ('AM62AX')
 
-   The macro that specifies the CMA size is CONFIG_CMA_SIZE_MBYTES present in the file :file:`arch/arm64/configs/tisdk_am62axx-evm_defconfig` in the linux directory of sdk. The default value is 576MB.
+   The CMA size can be adjusted by opening the board dts file :file:`arch/arm64/boot/dts/ti/k3-am62a7-sk.dts`,
+   finding the "linux,cma" node and changing the size attribute. The default value is 576MB.
 
-.. ifconfig:: CONFIG_part_variant in ('J784S4','J742S2')
+.. ifconfig:: CONFIG_part_variant in ('J784S4')
 
-   The macro that specifies the CMA size is CONFIG_CMA_SIZE_MBYTES present in the file :file:`arch/arm64/configs/tisdk_am62axx-evm_defconfig` in the linux directory of sdk. The default value is 1792MB.
+   The CMA size can be adjusted by opening the board dts file :file:`arch/arm64/boot/dts/ti/k3-j784s4-evm.dts`,
+   finding the "linux,cma" node and changing the size attribute. The default value is 1792MB.
 
+.. ifconfig:: CONFIG_part_variant in ('J722S')
+
+   The CMA size can be adjusted by opening the board dts file :file:`arch/arm64/boot/dts/ti/k3-j722s-evm.dts`,
+   finding the "linux,cma" node and changing the size attribute. The default value is 896MB.
+
+.. ifconfig:: CONFIG_part_variant in ('AM62PX')
+
+   The CMA size can be adjusted by opening the board dts file :file:`arch/arm64/boot/dts/ti/k3-am62p5-sk.dts`,
+   finding the "linux,cma" node and changing the size attribute. The default value is 576MB.
 
 The value can be increased according to the availability of space in DDR memory map.
-Also to change cma without re-compilation, one can stop at u-boot prompt during bootup and update cma as below and then boot:
+CMA allocation can also be changed without re-compilation. One can stop at u-boot prompt during boot and update the CMA parameter as shown below:
 
 .. code-block:: console
 
    target # setenv args_all $args_all cma=1000M
```

Comment:
Does this allocate buffers in lower or higher 32bit address ? Can we move to higher 32bit value, considering constraints in lower mem regions ?

Reply:
I have noticed that whenever change is made in kernel config, or u-boot level, memory allocation is always going in lower 32-bit and I have not seen a way to force it into higher mem. Only when allocation comes from device tree file does it allocate in higher mem.

Reply:
Yeah, this is kind of a big deal. It's something I've noticed as well, but yet to have found any documentation really explaining the discrepancy.